### PR TITLE
Fix panic in forc-fmt with special chars

### PR DESCRIPTION
**Pull request by sdankel**
## Description

Closes https://github.com/FuelLabs/sway/issues/5013

## Checklist

- [x] I have linked to any relevant issues.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation where relevant (API docs, the reference, and the Sway book).
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added (or requested a maintainer to add) the necessary `Breaking*` or `New Feature` labels where relevant.
- [ ] I have done my best to ensure that my PR adheres to [the Fuel Labs Code Review Standards](https://github.com/FuelLabs/rfcs/blob/master/text/code-standards/external-contributors.md).
- [ ] I have requested a review from the relevant team or maintainers.

### CODE REVIEW

File: swayfmt/src/utils/map/newline.rs [hunk 1]
```diff
@@ -278,7 +278,7 @@ fn insert_after_span(
         format_newline_sequence(&newline_sequence, threshold)
     )?;
     let mut src_rope = Rope::from_str(formatted_code);
-    src_rope.insert(at, &sequence_string);
+    src_rope.try_insert(at, &sequence_string).unwrap();
```
Comment (JoshuaBatty) [inline swayfmt/src/utils/map/newline.rs:281]: should we return an error here instead of unwrapping?

**Comment by JoshuaBatty** [comment]
nice, just a small clippy error otherwise happy to approve.

**Review by JoshuaBatty — DISMISSED** [review]

**Review by JoshuaBatty — APPROVED** [review]

File: swayfmt/src/utils/map/newline.rs [hunk 2]
```diff
@@ -278,7 +278,11 @@ fn insert_after_span(
         format_newline_sequence(&newline_sequence, threshold)
     )?;
     let mut src_rope = Rope::from_str(formatted_code);
-    src_rope.insert(at, &sequence_string);
+
+    if src_rope.try_insert(at, &sequence_string).is_err() {
+        return Err(FormatterError::NewlineSequenceError);
+    }
```
Comment (kayagokalp) [inline swayfmt/src/utils/map/newline.rs:284]: Also possible to write this in a single line, if you prefer:
```suggestion
    src_rope.try_insert(at, &sequence_string).map_err(|_| FormatterError::NewlineSequenceError)?;
```

**Review by kayagokalp — APPROVED** [review]
Thanks for handling this! LGTM :) left a single nit but more like a preference thing so nothing to block. Feel free to discard that comment

File: swayfmt/src/utils/map/newline.rs (hunk 2 above)
Comment (sdankel) [reply swayfmt/src/utils/map/newline.rs:284]: Ah yeah, that's what I wanted to do! :)

**Review by kayagokalp — APPROVED** [review]

**Review by JoshuaBatty — APPROVED** [review]

**Pull request merged**